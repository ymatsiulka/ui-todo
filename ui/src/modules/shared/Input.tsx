import React, { type InputHTMLAttributes } from 'react';
import styles from './Input.m.scss';

interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  name: string;
  label: string;
  labelClassName?: string;
}

const Input: React.FC<InputProps> = ({ name, label, children, ...rest }) => {
  return (
    <div className={styles.inputWrapper}>
      {children}

      <input id={name} {...rest} />
      <label htmlFor={name}>{label}</label>
    </div>
  );
};

export default Input;
