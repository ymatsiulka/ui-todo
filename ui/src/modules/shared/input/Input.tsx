import React, { InputHTMLAttributes } from 'react';

interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  name: string;
  label: string;
  labelClassName?: string;
}

const Input: React.FC<InputProps> = ({ name, label, children, ...rest }) => {
  return (
    <div className="input-wrapper">
      {children}

      <input id={name} {...rest} />
      {(label || name) && <label htmlFor={name}>{label}</label>}
    </div>
  );
};

export default Input;
