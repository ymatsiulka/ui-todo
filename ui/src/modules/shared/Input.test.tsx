import React from 'react';
import { render } from '@testing-library/react';
import Input from './Input';

test('Input should render properly', async () => {
  // Arrange
  // Act
  const { asFragment } = render(
    <Input name="input" label="label">
      Some text
    </Input>,
  );

  // Assert
  expect(asFragment()).toMatchSnapshot();
});
