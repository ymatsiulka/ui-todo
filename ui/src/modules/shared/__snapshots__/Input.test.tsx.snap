// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render properly 1`] = `
<DocumentFragment>
  <div
    class="inputWrapper"
  >
    Some text
    <input
      id="input"
      title="input"
    />
    <label
      for="input"
    >
      label
    </label>
  </div>
</DocumentFragment>
`;

exports[`Input when name and label is empty should render properly 1`] = `
<DocumentFragment>
  <div
    class="inputWrapper"
  >
    Some text
    <input
      id=""
      title=""
    />
  </div>
</DocumentFragment>
`;
